pkg/supervisor/prochandle_linux.go

```diff
@@ -101,6 +102,28 @@ func (p *unixProcess) requestGracefulTermination() error {
 	}
 }
 
+// awaitTermination implements [procHandle].
+func (p *unixProcess) awaitTermination(ctx context.Context) (err error) {
+	// The /proc/<pid> dirfds are not pollable. Use a pidfd, if available. This
+	// is the only way that doesn't involve userspace polling using timeouts and
+	// such, but requires at least Linux 5.3.
```

Comment:
Does this mean it's a hard requirement for k0s now?

Reply:
No, this should return `ENOSYS` if the syscall is unavailable, which in turn should trigger the `ErrUnsupported` fallback [here](https://github.com/k0sproject/k0s/pull/6418/files#diff-db2b621cbd5148e2f3977108fbed3b3a703b75bcc9a5fb77a42dd30e5d0ceb05R91-R97).